core/rawdb/bor_receipts.go

```diff
@@ -0,0 +1,192 @@
+package rawdb
+
+import (
+	"math/big"
+
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon/common"
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/rlp"
+	"github.com/ledgerwatch/log/v3"
+)
+
+var (
+	// bor receipt key
+	borReceiptKey = types.BorReceiptKey
+
+	// borTxLookupPrefix + hash -> transaction/receipt lookup metadata
+	borTxLookupPrefix = []byte("matic-bor-tx-lookup-")
+)
+
+// borTxLookupKey = borTxLookupPrefix + bor tx hash
+func borTxLookupKey(hash common.Hash) []byte {
+	return append(borTxLookupPrefix, hash.Bytes()...)
+}
+
+// HasBorReceipt verifies the existence of all block receipt belonging
+// to a block.
+func HasBorReceipts(db kv.Has, hash common.Hash, number uint64) bool {
+	if has, err := db.Has(kv.BorReceipts, borReceiptKey(number, hash)); !has || err != nil {
```

Comment:
Why need to create new table `kv.BorReceipts`? Why don't use existing `kv.Receipts`? ( same for other tables)

Reply:
Bor uses a separate receipts table because it stores the normal receipts in receipts and its own bor receipts for state syncs (kv.Receipts is for chain receipts, the other is for state receipts)